src/hazelcore/semantics/UHPat.re

```diff
@@ -153,6 +156,35 @@ and make_inconsistent_operand =
     (Parenthesized(set_p), u_gen);
   };
 
+let patterns_of_type =
+    (u_gen: MetaVarGen.t, ty: HTyp.t): (list(t), MetaVarGen.t) => {
+  let (hole1, u_gen) = u_gen |> new_EmptyHole;
+  let (hole2, u_gen) = u_gen |> new_EmptyHole;
+  let (hole3, u_gen) = u_gen |> new_EmptyHole;
+  let (hole4, u_gen) = u_gen |> new_EmptyHole;
```

Comment:
why are you generating 4 fresh holes in all situations?

Reply:
I wasn't sure if it mattered if generated holes aren't used. I can make another `switch` that generated the right number of holes as an array, but this feels more complicated.
Please tell me if you know a nice solution :)

Reply:
This happened to fix itself when I was working on generating the right number of holes for a product type, because all cases get their holes differently now